variables.tf

```diff
@@ -229,3 +198,58 @@ variable "manager_pod_annotations" {
   default     = {}
 }
 
+variable "additional_secrets" {
+  description = "additional secrets to mount into the manager pods"
+  type        = list(map(string))
+  default     = []
+}
+
+variable "replicas" {
+  description = "the number of manager pods to create"
+  type        = number
+  default     = 1
+}
+
+variable "runner_name" {
+  description = "name of the runner"
+  type        = string
+}
+
+variable "unregister_runners" {
+  description = "whether runners should be unregistered when pool is deprovisioned"
+  type        = bool
+  default     = true
+}
+
+variable "runner_token" {
+  description = "token of already registered runer. to use this var.runner_registration_token must be set to null"
+  type        = string
+  default     = null
+}
+
+
+variable "cache" {
+  description = "Describes the properties of the cache. type can be either of ['local', 'gcs', 's3', 'azure'], path defines a path to append to the bucket url, shared specifies whether the cache can be shared between runners. you also specify the individual properties of the particular cache type you select. see https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnerscache-section"
+  type = object({
+    type   = string
+    path   = string
+    shared = bool
+    gcs    = map(any)
+    s3     = map(any)
+    azure  = map(any)
+  })
+
+  validation {
```

Comment:
We should add validation for `azure` and `s3` cache type as well. Also add a validation that checks that `type` is only one of `gcs`, `azure`, `s3` or `local`

Reply:
for s3, ill just check if the s3 map is empty. because the values there can have any number of combinations.